### PR TITLE
LP-1812 Fix - don't include nationality when not changed in update person

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-1812

### Change description
Nationality was being included in filing data for Change partner person forms (LPCH03/04?) when nationalities had not changed.
For example, if person had no nationality2 and it was not changed/updated but name was, it would still include the nationalities in the filings data and then get sent to chips as part of the changed/updated data.

This pull request refines the logic for detecting changes to partner nationalities in the `FilingsService` and significantly expands the test coverage for this logic. The main updates include correcting the comparison logic for nationality fields and adding comprehensive unit tests to verify all scenarios, including case insensitivity and handling of null values.

**Logic improvements:**

* The logic for determining whether `nationality1` and `nationality2` have changed in `setNationalitiesFields` was corrected to handle null values and case-insensitive comparisons more accurately. This ensures updates are only triggered when there is a genuine change between the partner data and the appointment record.

**Test coverage enhancements:**

* Added a comprehensive set of unit tests in `FilingsServiceTest` that cover all combinations of null and non-null values for `nationality1` and `nationality2`, as well as case-insensitive comparisons and scenarios where nationalities are added, removed, or changed. These tests include a truth table for clarity.
* Refactored test helpers to allow injection of appointment nationality values, enabling more granular testing of the nationality change logic. [[1]](diffhunk://#diff-887f8fd6b484875468be6a7b1d3a9da67da2de6ef0e445a5a45b4376f75984ceR1160-R1171) [[2]](diffhunk://#diff-887f8fd6b484875468be6a7b1d3a9da67da2de6ef0e445a5a45b4376f75984ceL944-R1182)

These changes improve both the robustness of the nationality update detection and the reliability of the codebase by ensuring all edge cases are tested.

### Work checklist

* [x] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__
### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/filings/FilingsServiceTest.java [hunk 1]
```diff
@@ -777,6 +777,240 @@ void testFilingGenerationSuccessfulForRemovingLimitedPartnerPerson(
...
+         * Truth table:
+         * | partnerNationality1 | appointmentNationality1 | Expected nationality1Changed |
+         * |---------------------|-------------------------|------------------------------|
+         * | null                | null                    | false (unchanged)            |
+         * | null                | "British"               | true (changed)               |
+         * | "British"           | null                    | true (changed)               |
+         * | "British"           | "British"               | false (unchanged)            |
+         * | "British"           | "british" (diff case)   | false (unchanged)            |
+         * | "British"           | "French"                | true (changed)               |
+         */
```
Comment (leafdevelop-ch) [inline src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/filings/FilingsServiceTest.java:786]: Can this be change to a parameterised test that matches the conditions of this table?

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/filings/FilingsServiceTest.java:786]: tests now combined and parameterised

Comment (leafdevelop-ch) [reply src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/filings/FilingsServiceTest.java:786]: Nice

**Review by leafdevelop-ch — APPROVED** [review]
